### ISSUE TITLE
Blazor can not compile generic-typed component with generic array parameter

### ISSUE THREAD

**Issue by xiety** [dotnet/aspnetcore#15014]
### Describe the bug
Compiler error when using generic array parameter.

If there is no type parameter:

    error RZ10001: The type of component 'Component' cannot be inferred based on the values provided. Consider specifying the type arguments directly using the following attributes: 'TItem'.

If type parameter is specified:

    error CS0246: The type or namespace name 'TItem' could not be found (are you missing a using directive or an assembly reference?)

### To Reproduce
Component.razor

    @typeparam TItem
    @code {
        [Parameter]
        public TItem[] List { get; set; }
    }

Page.razor

    <Component TItem="System.Int32" List="new int[] { 1, 2, 3 }" />

### Expected behavior
Compile without errors.

### Actual behavior
Compilation error CS0246 in generated file:

    __builder.AddAttribute(1, "List", Microsoft.AspNetCore.Components.CompilerServices.RuntimeHelpers.TypeCheck<TItem[]>());

### Workaround

    IReadOnlyList<TItem> works fine

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @xiety.
We will look into this post 3.1 milestone. Your workaround looks good for now.

**Comment by moataz1991** [comment]
i have a question
can do something like this...

```
<input @bind="Value"/>

@code{
          [Parameter]
          public object Value {get; set;}
}
```

**Comment by mkArtakMSFT** [comment]
@rynowak is this by-design?

**Comment by SommerEngineering** [comment]
Would be great if all those generic issues in Blazor would be fixed soon. It's a sad thing that these bugs will probably remain in .NET 5.0 RTM. Most of the issues are tagged as `affected-very-few` 😞

**Comment by koryphaee** [comment]
I have the same issue in .NET 5.

**Comment by danarcher** [comment]
I too am one of the very few, .NET 5. It didn't even occur to me that Blazor wouldn't support TItem[] parameters on generic components; googling it was just a sanity check, I didn't expect to end up here.

**Comment by taylorchasewhite** [comment]
I also did not expect to end up here...

**Comment by mkArtakMSFT** [comment]
This is being fixed by dotnet/aspnetcore#34737

**Comment by SteveSandersonMS** [comment]
Closing as duplicate of https://github.com/dotnet/aspnetcore/issues/29105, which contains more info about the cause.

As mentioned by @mkArtakMSFT, it's being fixed in either https://github.com/dotnet/aspnetcore/pull/34737 or https://github.com/dotnet/aspnetcore/pull/34734
